lib/bundle/brewfile.rb

```diff
@@ -2,29 +2,39 @@
 
 module Bundle
   module Brewfile
+    BUNDLE_FILE_ENV_KEY = "HOMEBREW_BUNDLE_FILE"
     module_function
 
     def path(dash_writes_to_stdout: false)
-      if ARGV.include?("--global")
-        Pathname.new("#{ENV["HOME"]}/.Brewfile")
-      else
-        filename = ARGV.value("file")
-        if filename == "-"
-          filename = if dash_writes_to_stdout
-            "/dev/stdout"
+      env_bundle_file = ENV[BUNDLE_FILE_ENV_KEY]
+
+      filename =
+          if ARGV.include?("--global")
+            raise "'#{BUNDLE_FILE_ENV_KEY}' can not be specified with '--global'" unless env_bundle_file.blank?
+            "#{ENV["HOME"]}/.Brewfile"
+          elsif ARGV.include?("--file")
+            handle_file_value(ARGV.value("file"), dash_writes_to_stdout)
+          elsif env_bundle_file.present?
+            env_bundle_file
           else
-            "/dev/stdin"
+            "Brewfile"
           end
-        end
-        filename ||= "Brewfile"
-        Pathname.new(filename).expand_path(Dir.pwd)
-      end
+
+      Pathname.new(filename).expand_path(Dir.pwd)
     end
 
     def read
       Brewfile.path.read
     rescue Errno::ENOENT
       raise "No Brewfile found"
     end
+
+    def handle_file_value(filename, dash_writes_to_stdout)
+      if filename == "-"
```

Comment:
`if filename != "-"` and then could do `elsif dash_writes_to_stdout` and `else` which would make the function a little more readable.